### PR TITLE
#1379 -  Braintree Source incremental Sync 

### PR DESCRIPTION
**Pull request by yevhenii-ldv**
## What
*Implementation of incremental sync mode fore Braintree Singer Source*

## Pre-merge Checklist
- [ ] *Run integration tests*
- [ ] *Publish Docker images*

## Recommended reading order
1. `test.java`
1. `component.ts`
1. the rest

### CODE REVIEW

File: airbyte-integrations/connectors/source-braintree-singer/sample_files/configured_catalog.json [hunk 1]
```diff
@@ -193,8 +193,12 @@
...
           }
         },
-        "supported_sync_modes": ["full_refresh"]
-      }
+        "source_defined_cursor": true,
+        "supported_sync_modes": ["incremental"],
+        "default_cursor_field": []
+      },
+      "sync_mode": "incremental",
+      "cursor_field": []
```
Comment (vitaliizazmic) [inline airbyte-integrations/connectors/source-braintree-singer/sample_files/configured_catalog.json:201]: Why "cursor_field" and "default_cursor_field" are empty?

Comment (sherifnada) [reply airbyte-integrations/connectors/source-braintree-singer/sample_files/configured_catalog.json:201]: my guess is that the cursor field is not an actual field in the returned stream?

Comment (yevhenii-ldv) [reply airbyte-integrations/connectors/source-braintree-singer/sample_files/configured_catalog.json]: Yes, you are right, I have already correct this mistake

**Review by sherifnada — APPROVED** [review]

**Pull request merged**